lib/src/Uncertainty/Distribution/CombinationsDistribution.cxx

```diff
@@ -0,0 +1,518 @@
+Bool CombinationsDistribution::isIntegral() const
+{
+  return true;
+}
+
+/* Parameters value and description accessor */
+CombinationsDistribution::PointWithDescriptionCollection CombinationsDistribution::getParametersCollection() const
+{
+  const UnsignedInteger dimension = getDimension();
+  PointWithDescriptionCollection parameters((dimension == 1 ? 1 : dimension + 1));
+  for (UnsignedInteger i = 0; i < dimension; ++i)
+  {
+    PointWithDescription point({static_cast<Scalar>(n_)});
+    point.setDescription({"n"});
+    point.setName(getDescription()[i]);
+    parameters[i] = point;
+  }
+  if (dimension > 1)
+  {
+    PointWithDescription point(getParameter());
+    point.setDescription(getParameterDescription());
+    point.setName("dependence");
+    parameters[dimension] = point;
+  }
+  return parameters;
+}
+
+/* Parameters value accessor */
+Point CombinationsDistribution::getParameter() const
+{
+  return {static_cast<Scalar>(k_), static_cast<Scalar>(n_)};
+}
+
+void CombinationsDistribution::setParameter(const Point & parameter)
+{
+  if (parameter.getSize() != 2) throw InvalidArgumentException(HERE) << "CombinationsDistribution expected 2 parameters, got " << parameter.getSize();
+  if (parameter[0] != std::round(parameter[0])) throw InvalidArgumentException(HERE) << "CombinationsDistribution first parameter k must be an integer, got " << parameter[0];
+  if (parameter[1] != std::round(parameter[1])) throw InvalidArgumentException(HERE) << "CombinationsDistribution second parameter n must be an integer, got " << parameter[1];
+  const Scalar w = getWeight();
+  *this = CombinationsDistribution(parameter[0], parameter[1]);
+  setWeight(w);
+}
+
+/* Parameters description accessor */
+Description CombinationsDistribution::getParameterDescription() const
+{
+  return {"k", "n"};
+}
+
+/* K accessor */
+void CombinationsDistribution::setK(const UnsignedInteger k)
+{
+  if (k == 0) throw InvalidArgumentException(HERE) << "Error: k must be > 0.";
+  if (k > n_) throw InvalidArgumentException(HERE) << "Error: k must be less or equal to n, here k=" << k << " and n=" << n_;
+  if (k != k_)
+  {
+    k_ = k;
+    logPDFValue_ = SpecFunc::LogGamma(n_ - k_ + 1) - SpecFunc::LogGamma(n_ + 1) + SpecFunc::LogGamma(k_ + 1);
+    setDimension(k);
+    isAlreadyComputedMean_ = false;
+    isAlreadyComputedCovariance_ = false;
+    isAlreadyCreatedGeneratingFunction_ = false;
+  }
+}
+
+/* K accessor */
+UnsignedInteger CombinationsDistribution::getK() const
+{
+  return k_;
+}
+
+/* N accessor */
+void CombinationsDistribution::setN(const UnsignedInteger n)
+{
+  if (n == 0) throw InvalidArgumentException(HERE) << "Error: n must be > 0.";
+  if (n < k_) throw InvalidArgumentException(HERE) << "Error: n must be greater or equal to k, here n=" << n << " and k=" << k_;
+  if (n != n_)
+  {
+    n_ = n;
+    logPDFValue_ = SpecFunc::LogGamma(n_ - k_ + 1) - SpecFunc::LogGamma(n_ + 1) + SpecFunc::LogGamma(k_ + 1);
```

Comment:
Is seems that this is the opposite of the logarithm of the binomial coefficient. Isn't this the purpose of the newly created `SpecFunc::LogBinomialCoefficient` method?

Reply:
not exactly

Reply:
I cannot see the difference: can you elaborate?

Reply:
its not exactly -LogBinomialCoefficient(n+1,k+1) as the first term is loggamma(n-k+1) instead loggamma(n-k)

Reply:
I cannot see why from the code. Isn'it consistent with the definition of the distribution:

![image](https://github.com/user-attachments/assets/efac8443-aabb-4a65-ab64-c1cca7b44b23)

This implies:

$$
\log\left(\mathbb{P}(\boldsymbol{X} = \boldsymbol{x})\right) = \log(1) - \log\left(\binom{n}{k}\right)
$$

and since $\log(1) = 0$, then:

$$
\log\left(\mathbb{P}(\boldsymbol{X} = \boldsymbol{x})\right) = - \log\left(\binom{n}{k}\right).
$$

This is why the right hand side must be `-LogBinomialCoefficient(n,k)`, isn't it?